airflow_prometheus_exporter/prometheus_exporter.py

```diff
@@ -381,30 +383,62 @@ def get_num_queued_tasks():
         )
 
 
+def sla_check(sla_interval, sla_time, max_execution_date, cadence, execution_dates):
+    utc_datetime = pytz.timezone("UTC").localize(datetime.datetime.utcnow())
+    if sla_time:
+        local_datetime = utc_datetime.astimezone(pytz.timezone(TIMEZONE_LA))
+        sla_datetime = pytz.timezone(TIMEZONE_LA).localize(
+            datetime.datetime.combine(
+                local_datetime.date(), datetime.time(int(sla_time))
+            )
+        )
+    else:
+        sla_datetime = utc_datetime
+
+    interval_in_second = parse(sla_interval)
+    checkpoint = sla_datetime - datetime.timedelta(seconds=interval_in_second)
+    if utc_datetime >= sla_datetime and max_execution_date < checkpoint:
+        return True
+
+    if cadence != "triggered":
+        # Check the state of previous run before sla_time.
+        # To detect consecutive failed scenario.
+        # Filter out triggered DAGs e.g. PPD
+        for record in execution_dates:
```

Comment:
i think this control logic is a little strange. unless there are a lot of execution_dates to look through, you could probably just leave this loop as
```
for record in execution_dates:
    if record['execution_date'] <= checkpoint:
        if record['state'] != 'success':
            return True
```

Reply:
Sure. 

I was thinking of letting the final False to handle sla no miss. In case we add other conditions. It's updated now.

Reply:
Updated